backend/tests/external/aws/test_iam.py

```diff
@@ -0,0 +1,15 @@
+from src.external.aws.iam import Iam
+
+
+def test_generate_role_policy_with_allow_works():
+    principle = "user"
+    effect = "allow"
+    arn = "mytestarn/testing/*"
+
+    role = Iam._generate_role_policy(principle, effect, arn)
+    assert role is not None
+    assert isinstance(role, dict)
+    assert isinstance(role["policyDocument"]["Statement"], list)
+    assert role["policyDocument"]["Statement"][0]["Action"] == "execute-api:Invoke"
+    assert role["policyDocument"]["Statement"][0]["Effect"] == "allow"
```

Comment:
You can make the test less brittle by connecting the input with your assertion.  You may even be able to do the same below for the `arn`.

```suggestion
    assert role["policyDocument"]["Statement"][0]["Effect"] == effect
```